pkg/storage/chunk/cache/groupcache.go

```diff
@@ -92,6 +93,10 @@ func NewGroupCache(rm ringManager, config GroupCacheConfig, logger log.Logger, r
 	addr := fmt.Sprintf("http://%s", rm.Addr())
 	level.Info(logger).Log("msg", "groupcache local address set to", "addr", addr)
 
+	http2Transport.ReadIdleTimeout = config.HeartbeatInterval
+	http2Transport.PingTimeout = config.HeartbeatTimeout
+	http2Transport.WriteByteTimeout = config.WriteByteTimeout
```

Comment:
Which of these would affect the read timeout?

Reply:
`ReadIdleTimeout` and `PingTimout` combined form a sort of "read timeout".

`ReadIdleTimeout` sends a heartbeat when the connection goes idle. This ensures that we test the connection at least once every `ReadIdleTimeout`. 

`PingTimeout` is our tolerance for heartbeat responses.

Lets assume our connection has hung, `ReadIdleTimeout` is 1s, and `PingTimout` is 100ms:
1. No traffic occurs in 1s so we send a HeartBeat
2. No `ACK` is received in 100ms, terminate the connection

In the case where traffic is flowing normally, the connection doesn't send heartbeats because `ReadIdleTimeout` is never reached.